AutomationHelpers/test/FileLibraryTests.cs

```diff
@@ -0,0 +1,426 @@
+        {
+            //Arrange
+            var logger = new TestReportLogger();
+            Report.AttachLogger(logger);
+            var filePath1 = Path.Combine(testFilesDirPath, "TextFileUnixEOL.txt");
+            var filePath2 = Path.Combine(testFilesDirPath, "TextFileWinEOL.txt");
+
+            //Act
+            FileLibrary.ValidateFilesTextEqual(filePath1, filePath2, true);
+
+            //Assert
+            Report.DetachLogger(logger);
+            Assert.AreEqual(
+                string.Format("Files '{0}' and '{1}' are equal.", filePath1, filePath2),
+                logger.LastLogMessage);
+        }
+
+        [Test]
+        public void ValidateFileContainsText_SearchTextInNonexistentFile_ReportFileNotExist()
+        {
+            //Arrange
+            var logger = new TestReportLogger();
+            Report.AttachLogger(logger);
+            var filePath = Path.Combine(testFilesDirPath, "TextFileForSearch1.txt");
+            var textToSearchFor = "Maecenas tristique consequat est, et condimentum";
+
+            //Act
+            FileLibrary.ValidateFileContainsText(filePath, textToSearchFor);
+
+            //Assert
+            Report.DetachLogger(logger);
+            Assert.AreEqual(
+                string.Format("The file '{0}' does not exist.", filePath),
+                logger.LastLogMessage);
+        }
+
+        [Test]
+        public void ValidateFileContainsText_SearchText_ReportTextNotFound()
+        {
+            //Arrange
+            var logger = new TestReportLogger();
+            Report.AttachLogger(logger);
+            var filePath = Path.Combine(testFilesDirPath, "TextFileForSearch.txt");
+            var textToSearchFor = "Maecenas123 tristique consequat est, et condimentum";
+
+            //Act
+            FileLibrary.ValidateFileContainsText(filePath, textToSearchFor);
+
+            //Assert
+            Report.DetachLogger(logger);
+            Assert.AreEqual(
+                string.Format("Text '{0}' was not found in file '{1}'.", textToSearchFor, filePath),
+                logger.LastLogMessage);
+        }
+
+        [Test]
+        public void ValidateFileContainsText_SearchText_ReportTextFound()
+        {
+            //Arrange
+            var logger = new TestReportLogger();
+            Report.AttachLogger(logger);
+            var filePath = Path.Combine(testFilesDirPath, "TextFileForSearch.txt");
+            var textToSearchFor = "Maecenas tristique consequat est, et condimentum";
+
+            //Act
+            FileLibrary.ValidateFileContainsText(filePath, textToSearchFor);
+
+            //Assert
+            Report.DetachLogger(logger);
+            Assert.AreEqual(
+                string.Format("Text '{0}' was found on line 7: 'Maecenas tristique consequat est, et condimentum lacus bibendum non. Nam semper malesuada risus ut interdum.'.", textToSearchFor),
+                logger.LastLogMessage);
+        }
+
+        /*
```

Comment:
Probably don't leave that in there for the final version :-)

Reply:
Removed. :)